notebooks/00_preprocess_data.ipynb

```diff
@@ -0,0 +1,494 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot coordinates\n",
+    "sql = \"\"\"\n",
+    "    -- Get all plot records for full-plot forested conditions\n",
+    "    -- between 2011 and 2020\n",
+    "    select\n",
+    "      p.CN as PLT_CN,\n",
+    "      p.MEASYEAR as ASSESSMENT_YEAR,\n",
+    "      p.LON,\n",
+    "      p.LAT\n",
+    "    from\n",
+    "      plot p\n",
+    "    join\n",
+    "      cond c on p.CN = c.PLT_CN\n",
+    "    where\n",
+    "      c.CONDPROP_UNADJ = 1.0\n",
+    "      and c.COND_STATUS_CD = 1\n",
+    "      and p.INVYR >= 2011\n",
+    "      and p.INVYR <= 2020\n",
+    "    order by p.CN\n",
+    "\"\"\"\n",
+    "plot_df = pd.read_sql_query(sql, connection)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This produces a pandas `DataFrame` with n=5,671 records and 4 columns."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(plot_df.head())\n",
+    "print(f\"Plot table shape: {plot_df.shape}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "At this point, we are done with the database and can close the connection."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "connection.close()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Step 4: Transform the data to deal with unit conversions, species codes, and projected coordinates"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For use in `synthetic-knn`, we make the following assumptions about the input tree data:\n",
+    "\n",
+    "- Units are metric (e.g. diameter in cm, density in trees per hectare)\n",
+    "- Species are represented by USDA PLANTS database codes\n",
+    "- Unique tree and plot IDs are stored as integers\n",
+    "\n",
+    "In order to crosswalk FIA species codes to USDA PLANTS database codes, we use the `FIA_TO_PLANTS_XWALK` dictionary from the `species_crosswalk` module in this directory. We also need to transform the geographic coordinates from FIADB to the reference projection of our spatial data (USGS National Albers).  We will modify the `tree_df` and `plot_df` dataframes in place to make these changes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xwalk_df = pd.read_csv(\"./data/REF_SPECIES.csv\")\n",
+    "FIA_TO_PLANTS_XWALK = pd.Series(xwalk_df.SPCD.values, index=xwalk_df.FIA_SYMBOL).to_dict()\n",
```

Comment:
I'm getting `AttributeError: 'DataFrame' object has no attribute 'FIA_SYMBOL'`, and it doesn't appear to be included in the CSV

Reply:
Sorry, this was me changing up the notebook and not testing it again.  I had originally had a separate module that read from the `REF_SPECIES.csv` table and I imported that module.  That seemed like overkill, so I just did it directly in the notebook and obviously hadn't tested (especially because I swapped the keys/values!)  The field should have been called `SPECIES_SYMBOL`.

I've also linked the source data for the `REF_SPECIES.csv` table.